dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/servertype/internal/LibertyMavenJEEPublisher.java

```diff
@@ -191,7 +191,9 @@ protected IStatus publishModule(int kind, PublishUnit publishUnit, IProgressMoni
 
                                     String serverDir = config.getConfigValue(ConfigurationType.serverDirectory);
                                     String appsDir = config.getConfigValue(ConfigurationType.appsDirectory);
-                                    String appName = config.getConfigValue(ConfigurationType.applicationFilename);
+                                    String version = config.getVersion();
+                                    String appName = config.getConfigValue(version != null
```

Comment:
This is repeated in quite a few places.  Can it be commoned up into a method?

Reply:
applicationFilename is now used.  I've reverted the changes to accomodate applicationFileName.